src/jdk.incubator.code/share/classes/jdk/incubator/code/internal/CodeModelToAST.java

```diff
@@ -0,0 +1,287 @@
+            JavaType jt = JavaType.type(javaBaseType.describeConstable().get());
+            Type t = syms.enterClass(syms.java_base, jt.externalize().toString());
+            m.put(jt, t);
+        }
+
+        m.putAll(Map.ofEntries(
+                Map.entry(JavaType.BOOLEAN, syms.booleanType),
+                Map.entry(JavaType.BYTE, syms.byteType),
+                Map.entry(JavaType.SHORT, syms.shortType),
+                Map.entry(JavaType.CHAR, syms.charType),
+                Map.entry(JavaType.INT, syms.intType),
+                Map.entry(JavaType.LONG, syms.longType),
+                Map.entry(JavaType.FLOAT, syms.floatType),
+                Map.entry(JavaType.DOUBLE, syms.doubleType)
+        ));
+
+        return m;
+    }
+
+    private Type typeElementToType(TypeElement te) {
+        JavaType jt = (JavaType) te;
+        return switch (jt) {
+            case ClassType ct when ct.hasTypeArguments() -> {
+                ClassType enclosingType = ct.enclosingType().orElse(null);
+                ListBuffer<Type> typeArgs = new ListBuffer<>();
+                for (JavaType typeArgument : ct.typeArguments()) {
+                    typeArgs.add(typeElementToType(typeArgument));
+                }
+                yield new Type.ClassType(typeElementToType(enclosingType), typeArgs.toList(),
+                        typeElementToType(ct.rawType()).tsym);
+            }
+            case ArrayType at -> {
+                yield new Type.ArrayType(typeElementToType(at.componentType()), syms.arrayClass);
+            }
+            case null -> Type.noType;
+            default -> {
+                if (!jtToType.containsKey(te)) {
+                    throw new IllegalStateException("JavaType -> Type not found for: " + te.externalize().toString());
+                }
+                yield jtToType.get(te);
+            }
+        };
+    }
+
+    private JCTree invokeOpToJCMethodInvocation(CoreOp.InvokeOp invokeOp) {
+        Method method;
+        try {
+            method = invokeOp.invokeDescriptor().resolveToDirectMethod(MethodHandles.lookup());
+        } catch (ReflectiveOperationException e) {
+            throw new RuntimeException(e);
+        }
+        long flags = method.getModifiers();
+        var name = names.fromString(invokeOp.invokeDescriptor().name());
+        Value receiver = invokeOp.invokeKind() == CoreOp.InvokeOp.InvokeKind.INSTANCE ? invokeOp.operands().get(0) : null;
+        List<Value> arguments = invokeOp.operands().stream().skip(receiver == null ? 0 : 1).collect(List.collector());
+        var paramTypes = invokeOp.invokeDescriptor().type().parameterTypes().stream().map(this::typeElementToType)
+                .collect(List.collector());
+        var restype = typeElementToType(invokeOp.resultType());
+        var type = new Type.MethodType(paramTypes, restype, List.nil(), syms.methodClass);
+        var methodSym = new Symbol.MethodSymbol(flags, name, type,
```

Comment:
Same as for field access (see below) -- we should not create a new method symbol from scratch but "resolve" a method in a given class symbol (using `Resolve::findInternalMethod`)

Reply:
This is related to the one above for var symbol -- creating method/var symbols directly in the backend is not the way to do things -- we should lean on `Resolve` instead. It would be nice to fix these -- but we can also take another pass at it later if required.